course7/homeworks/anton/dc/complex_stack.cc

```diff
@@ -0,0 +1,59 @@
+#include "complex_stack.h"
+
+bool digit_is_comming(std::istream& s)
+{
+    return '0' <= s.peek() && s.peek() <= '9';
+}
+
+bool number_is_comming(std::istream& s)
+{
+    return '_' == s.peek() || 'i' == s.peek() || digit_is_comming(s);
+}
+
+std::ostream& ComplexStack::operator>>(std::ostream& s)
+{
+    std::stack<complex::Complex> tmp_stack;
+    while(!stack.empty()) {
+        std::cout << stack.top() << std::endl;
+        tmp_stack.push(stack.top());
+        stack.pop();
+    }
+
+    while(!tmp_stack.empty()) {
+        stack.push(tmp_stack.top());
+        tmp_stack.pop();
+    }
+    return s;
+}
+
+bool ComplexStack::try_read_number(std::istream& s)
+{
+    if (number_is_comming(std::cin))
+    {
+        complex::Complex number;
+        std::cin >> number;
+        stack.push(number);
+        return true;
+    }
+    return false;
+}
+
+void ComplexStack::exec_operation(complex::operation op)
```

Comment:
This is completely unnecessary, right? There is no need for dynamic dispatch if you're coming from a static ```switch``` in main.
IMO, unneeded complexity is the worst of bugs.

Reply:
agree with you partially, it is not *completely* unnecessary since it makes this class a bit more extendable and reusable and allows to implement your next requirement of using DLs, right?